istioctl/cmd/analyze.go

```diff
@@ -160,6 +166,18 @@ istioctl analyze -L
 
 			sa := local.NewSourceAnalyzer(metadata.MustGet(), analyzers.AllCombined(), selectedNamespace, istioNamespace, nil, useDiscovery)
 
+			var suppressions []snapshotter.AnalysisSuppression
+			for _, s := range suppress {
+				parts := strings.Split(s, "=")
+				if len(parts) != 2 {
+					return fmt.Errorf("%s is not a valid suppression value. See istioctl analyze --help", s)
+				}
+				suppressions = append(suppressions, snapshotter.AnalysisSuppression{
+					Code:         parts[0],
```

Comment:
Should we validate the codes here? e.g. if I put in IST010BOGUS, right now it'll just not suppress anything. Do you think it's worth the effort to check against the list of valid codes?

Reply:
I kinda don't want to, mainly due to versioning problems. If a user downgrades istioctl (or we remove an old name for whatever reason) it would be kinda annoying if the command refused to run because the code no longer makes sense.

What do you think?

Reply:
sounds like we need an analyzer!

We could probably just output a warning though

Reply:
Yeah, I don't think the command needs to fail, but it would be nice to include a warning (even if it's just in the `--verbose` output). 

Definitely a "nice to have" rather than anything critical, though. If you think this PR is involved enough already we can create a follow-up task.

Reply:
It doesn't look like there's any easy way to enumerate the list of codes currently. I'll open another PR to support this.

Reply:
Once 19957 is in, I'll add it. Thanks for the suggestion overall!

Reply:
This is now added. The command emits a warning if the analyzer code is unknown.